pkg/keys/fulcio.go

```diff
@@ -0,0 +1,280 @@
+package keys
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/coreos/go-oidc"
+	httptransport "github.com/go-openapi/runtime/client"
+	fulcioClient "github.com/sigstore/fulcio/pkg/client"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/sigstore/fulcio/pkg/generated/client/operations"
+	"github.com/sigstore/fulcio/pkg/generated/models"
+	"github.com/sigstore/sigstore/pkg/oauthflow"
+	"github.com/theupdateframework/go-tuf/data"
+)
+
+func init() {
+	SignerMap.Store(data.KeyTypeSigstore_Fulcio_SHA256, NewFulcioSigner)
+	VerifierMap.Store(data.KeyTypeSigstore_Fulcio_SHA256, NewFulcioVerifier)
+}
+
+func NewFulcioSigner() Signer {
+	return &fulcioSigner{}
+}
+
+func NewFulcioVerifier() Verifier {
+	return &fulcioVerifier{}
+}
+
+type fulcioVerifier struct {
+	Identity string `json:"identity"`
+	Issuer   string `json:"issuer"`
+	key      *data.PublicKey
+}
+
+func (e *fulcioVerifier) Public() string {
+	return fmt.Sprintf("%s:%s", e.Issuer, e.Identity)
+}
+
+func (e *fulcioVerifier) Verify(msg, sig []byte) error {
+	// Check signature verification by extracting public key out of x509 cert
```

Comment:
As I'm sure you know, this needs to be implemented :)

Reply:
Added initial verification! Here's the missing pieces....
* Uploading and then verifying a rekor entry. 
* Verifying the cert chain to the Fulcio Root CA. Is this part pinned by the key scheme pointing to the fulcio "https://fulcio.sigstore.dev" effectively? Fulcio doesn't actually serve it's root certificate (https://github.com/sigstore/fulcio/blob/main/openapi.yaml), in cosign we rely on the TUF root. What do you think? I do think fulcio should serve it, but not sure if that's a risk.

Reply:
That kind of brings the question of, how do we "trust" rekor, or the timestamping service that will attest that the sig was created when the cert was valid? should that be pinned as well in root.json?

Should we also avoid the network call to the timestamping service by adding in a signed timestamp (by the trusted service) similar to how cosign does it?